### PR TITLE
Fix warnings in doc build due to ndcube 2.1.X

### LINKED ISSUE #1028: RTD admin please rebuild latest
https://readthedocs.org/projects/specutils/builds/

Recent PRs are failing with what seems unrelated errors. I have a feeling that if you rebuild the "latest" from current `main`, you would also see the same failure. At the time of posting, "latest" is 1.5 weeks old.

- [x] Need new ndcube release with sunpy/ndcube#600
- [x] https://github.com/sunpy/ndcube/issues/604

### PR DESCRIPTION
**Pull request by nabobalis**
With ndcube 2.1.2, we have activated nitpicky and fixed the build upstream.
With this PR, I update the docs config and ignore file to make RTD pass.

### EDIT

* https://github.com/sunpy/ndcube/issues/604
* fix #1028

### CODE REVIEW

File: docs/nitpick-exceptions [hunk 1]
```diff
@@ -15,28 +15,18 @@ py:obj SpectralQuantity
 py:obj SkyCoord
 
 # Super class for Spectrum1D for which we don't control the API links.
+py:class ndcube.mixins.ndslicing.NDCubeSlicingMixin
```
Comment (nabobalis) [inline docs/nitpick-exceptions:18]: I think most of this came from ndcube, so I recreated the list from scratch. Makes it shorter too.

**Comment by pllim** [comment]
Thanks! The nitpick changes looks fine as long as they make the warnings go away. We can remove the ignore if people ever complain about them links not working (I doubt people even read the docs most of the time so I don't think anyone ever gonna complain).

However, I think some upstream changes still needed. I guess they will go away when https://github.com/sunpy/ndcube/pull/607 is merged and released?

```
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.axis_world_coords:3: WARNING: py:obj reference target not found: ndcube.NDCube.world_axis_physical_types
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.axis_world_coords_values:5: WARNING: py:obj reference target not found: ndcube.NDCube.wcs.world_axis_physical_types
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.crop:35: WARNING: py:obj reference target not found: ndcube..ndcube.NDCubeABC
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.crop_by_values:1: WARNING: py:obj reference target not found: ndcube.NDCube.crop()
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.crop_by_values:39: WARNING: py:obj reference target not found: ndcube.ndcube.NDCubeABC
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCube.plot:1: WARNING: py:obj reference target not found: NDCube.plotter
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCube.rebin:28: WARNING: py:obj reference target not found: ndcube.utils.cube.propagate_rebin_uncertainty
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCube.rebin:51: WARNING: py:obj reference target not found: NDCube
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCube.rebin:99: WARNING: py:obj reference target not found: ndcube.NDCube.rebin
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.reproject_to:7: WARNING: py:obj reference target not found: 'interpolation'
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.reproject_to:7: WARNING: py:obj reference target not found: 'adaptive'
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.reproject_to:7: WARNING: py:obj reference target not found: 'exact'
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCubeBase.reproject_to:19: WARNING: py:obj reference target not found: True`
ndcube/ndcube.py:docstring of ndcube.ndcube.NDCube.to:25: WARNING: py:obj reference target not found: ǸDCube
```

**Comment by nabobalis** [comment]
They will all go when that 607 is merged. Leaving only the ones in the nitpick list left.

I hope to get that merged this week upstream.

**Comment by pllim** [comment]
Sounds good. Thanks!

**Comment by nabobalis** [comment]
I think this is now ready for a review. The docbuild is now passing with ndcube 2.1.2

**Review by pllim — APPROVED** [review]
Thank you!

I do not understand the difference between napoleon and numpydoc either. LGTM.

**Pull request merged**